### PR TITLE
Update fpdf2 stubs to 2.6.1

### PR DESCRIPTION
**Pull request by srittau**
Closes #9542

### CODE REVIEW

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 1]
```diff
@@ -144,6 +149,15 @@ class FPDF:
...
     ) -> None: ...
+    # The following definition crashes stubtest.
+    # def set_encryption(
+    #    self,
+    #    owner_password: str,
+    #    user_password: str | None = None,
+    #    encryption_method: EncryptionMethod | str = ...,
+    #    permissions: AccessPermission = ...,
+    #    encrypt_metadata: bool = False,
+    # ) -> None: ...
```
Comment (srittau) [inline stubs/fpdf2/fpdf/fpdf.pyi:160]: This needs further investigation for which I didn't have time yet.

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 2]
```diff
@@ -288,8 +300,12 @@ class FPDF:
     def set_font_size(self, size: float) -> None: ...
     def set_char_spacing(self, spacing: float) -> None: ...
     def set_stretching(self, stretching: float) -> None: ...
-    def add_link(self) -> int: ...
-    def set_link(self, link, y: int = ..., x: int = ..., page: int = ..., zoom: float | Literal["null"] = ...) -> None: ...
+    def add_link(
+        self, y: float = 0, x: float = 0, page: int = -1, zoom: float | Literal["null"] = "null"  # noqa: Y020
```
Comment (AlexWaygood) [inline stubs/fpdf2/fpdf/fpdf.pyi:304]: Can you file this as a bug over at flake8-pyi so we track it?

Comment (AlexWaygood) [reply stubs/fpdf2/fpdf/fpdf.pyi:304]: Oh wait it's already been fixed, hasn't it

Comment (AlexWaygood) [reply stubs/fpdf2/fpdf/fpdf.pyi:304]: Don't mind me

File: stubs/fpdf2/METADATA.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
-version = "2.6.0"
-requires = ["types-Pillow"]
+version = "2.6.1"
+requires = ["cryptography", "types-Pillow"]
```
Comment (AlexWaygood) [inline stubs/fpdf2/METADATA.toml:2]: Doesn't look like the `cryptography` dependency is being used anywhere

```suggestion
requires = ["types-Pillow"]
```

**Comment by AlexWaygood** [comment]
I think we'll need a pytype release that includes https://github.com/google/pytype/commit/829c5895f59d730cf8719d4dd6180364e40cdcb4 if you want to keep the string default values — or you could take them out for now, and add them back when we have the new pytype release

File: stubs/fpdf2/fpdf/fpdf.pyi (hunk 2 above)
Comment (AlexWaygood) [reply stubs/fpdf2/fpdf/fpdf.pyi:304]: Thanks! I forgot that I merged https://github.com/PyCQA/flake8-pyi/pull/318/ only last week, which I think already fixed this

File: stubs/fpdf2/METADATA.toml (hunk 3 above)
Comment (srittau) [reply stubs/fpdf2/METADATA.toml]: Moved to correct file.

File: stubs/fpdf2/fpdf/fpdf.pyi (hunk 1 above)
Comment (srittau) [reply stubs/fpdf2/fpdf/fpdf.pyi:158]: I just ran this against stubtest HEAD and this seems to fix it. I'll update the comment.

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 4]
```diff
@@ -415,14 +431,15 @@ class FPDF:
...
-        type: str = ...,
-        link: str = ...,
-        title: str | None = ...,
-        alt_text: str | None = ...,
+        x: float | Align | None = None,
+        y: float | None = None,
+        w: float = 0,
+        h: float = 0,
+        type: str = "",
+        link: str = "",
```
Comment (AlexWaygood) [inline stubs/fpdf2/fpdf/fpdf.pyi:439]:
```suggestion
        type: str = ...,
        link: str = ...,
```

File: stubs/fpdf2/fpdf/linearization.pyi [hunk 5]
```diff
@@ -34,7 +34,7 @@ class PDFXrefAndTrailer(ContentWithoutID):
     def is_first_xref(self) -> bool: ...
     @property
     def is_main_xref(self) -> bool: ...
-    def serialize(self) -> str: ...
+    def serialize(self, _security_handler: object = None) -> str: ...
```
Comment (AlexWaygood) [inline stubs/fpdf2/fpdf/linearization.pyi:37]: Should this be `StandardSecurityHandler | None` as well, like all the other `_security_handler` parameters?

Comment (srittau) [reply stubs/fpdf2/fpdf/linearization.pyi:37]: Ah yes, I knew that I missed one.

**Review by AlexWaygood — APPROVED** [review]

**Comment by rchen152** [comment]
> @rchen152 Is there a timeline for the next pytype release? This is non-critical, since we can easily work around the issue, but if a release is due soon, I would rather wait.

Sorry for the delay! (Yesterday was MLK day here.) We usually aim to do one pytype release a week. It's a manual process, so there's no fixed schedule. Yesterday's nightly tests were clean, so I can cut a release right now, actually.

**Comment by AlexWaygood** [comment]
> Yesterday's nightly tests were clean, so I can cut a release right now, actually.

That would be great, if possible! It would help unblock #9501

**Comment by rchen152** [comment]
Done! The new pytype version is 2023.01.17.

**Comment by AlexWaygood** [comment]
> Done! The new pytype version is 2023.01.17.

Amazing, thank you!

**Pull request merged**